### PR TITLE
S2259: Add FP reproducers from Peach

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.cs [hunk 1]
```diff
@@ -1897,3 +1897,88 @@ public void WithNormalValue()
...
     }
 }
+
+public class PeachReproducers
+{
+    public void ConditionalPropertyInt(string value)
+    {
+        if (value?.Length > 0)
+        {
+            value.ToString();   // Noncompliant
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.cs:1907]: Here and the follwing
```suggestion
            value.ToString();   // Noncompliant FP
```

**Review by Tim-Pohlmann — APPROVED** [review]
LGTM

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.cs (hunk 1 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.cs]: cc @Tim-Pohlmann: Don't trust me.. I forgot to put these "FP" comments all over the place 😅

Comment (Tim-Pohlmann) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.cs]: I noticed, but Martin had it pointed out already. I didn't think about the fact, that my approval would greenlight the PR for merging. Should I choose "Request Changes" and point to the unresolved comments in such a case?

Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.cs]: I saw the timeline and I supposed you know what you did. I just wanted to highlight it to be sure.

So as long as you trust me to address the comment, you did the right thing.

**Pull request merged**